stboot.go

```diff
@@ -235,25 +206,25 @@ func main() {
 	}
 
 	// Network interface
-	if securityConfig.BootMode == config.NetworkBoot {
-		switch hostConfig.IPAddrMode {
-		case config.StaticIP:
-			if err := network.ConfigureStatic(hostConfig); err != nil {
+	if stOptions.BootMode == opts.NetworkBoot {
+		switch stOptions.IPAddrMode {
+		case opts.IPStatic:
+			if err := network.ConfigureStatic(&stOptions.HostCfg); err != nil {
 				stlog.Error("cannot set up IO: %v", err)
 				host.Recover()
 			}
-		case config.DynamicIP:
-			if err := network.ConfigureDHCP(hostConfig, *doDebug); err != nil {
+		case opts.IPDynamic:
+			if err := network.ConfigureDHCP(&stOptions.HostCfg, *doDebug); err != nil {
 				stlog.Error("cannot set up IO: %v", err)
 				host.Recover()
 			}
 		default:
-			stlog.Error("unknown network mode: %s", hostConfig.IPAddrMode.String())
+			stlog.Error("unknown network mode: %s", stOptions.IPAddrMode.String())
```

Comment:
same here, no default required.

Reply:
Right. However, I would still go with a default to catch that invalid state, but I redo the message